From 50b181819224c717a468ced24ec7106e87b02dcd Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Tue, 28 Mar 2023 15:16:57 +0100
Subject: [PATCH] dtoverlays: Update compatible strings for
 imx290/327/462

Now upstream have accepted patches adding a mono version of
IMX290, and IMX327, update the compatible strings in the overlays
to match.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/imx290-overlay.dts | 4 ++--
 arch/arm/boot/dts/overlays/imx290_327.dtsi    | 2 +-
 arch/arm/boot/dts/overlays/imx327-overlay.dts | 5 +++--
 arch/arm/boot/dts/overlays/imx462-overlay.dts | 6 ++++--
 4 files changed, 10 insertions(+), 7 deletions(-)

--- a/arch/arm/boot/dts/overlays/imx290-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx290-overlay.dts
@@ -15,14 +15,14 @@
 	fragment@101 {
 		target = <&cam_node>;
 		__overlay__ {
-			compatible = "sony,imx290";
+			compatible = "sony,imx290lqr";
 		};
 	};
 
 	fragment@102 {
 		target = <&cam_node>;
 		__dormant__ {
-			compatible = "sony,imx290-mono";
+			compatible = "sony,imx290llr";
 		};
 	};
 
--- a/arch/arm/boot/dts/overlays/imx290_327.dtsi
+++ b/arch/arm/boot/dts/overlays/imx290_327.dtsi
@@ -1,7 +1,7 @@
 // Fragment to configure and IMX290 / IMX327 / IMX462 image sensor
 
 cam_node: imx290@1a {
-	compatible = "sony,imx290";
+	compatible = "sony,imx290lqr";
 	reg = <0x1a>;
 	status = "disabled";
 
--- a/arch/arm/boot/dts/overlays/imx327-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx327-overlay.dts
@@ -15,14 +15,15 @@
 	fragment@101 {
 		target = <&cam_node>;
 		__overlay__ {
-			compatible = "sony,imx327";
+			compatible = "sony,imx327lqr";
 		};
 	};
 
 	fragment@102 {
 		target = <&cam_node>;
 		__dormant__ {
-			compatible = "sony,imx327-mono";
+			// IMX327 mono is undefined in the binding - use imx290
+			compatible = "sony,imx290llr";
 		};
 	};
 
--- a/arch/arm/boot/dts/overlays/imx462-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx462-overlay.dts
@@ -12,17 +12,19 @@
 	// Fragment numbers deliberately high to avoid conflicts with the
 	// included imx290_327 overlay file.
 
+	//IMX462 is not defined in the bindings, so use IMX290 for now.
+
 	fragment@101 {
 		target = <&cam_node>;
 		__overlay__ {
-			compatible = "sony,imx462";
+			compatible = "sony,imx290lqr";
 		};
 	};
 
 	fragment@102 {
 		target = <&cam_node>;
 		__dormant__ {
-			compatible = "sony,imx462-mono";
+			compatible = "sony,imx290llr";
 		};
 	};
 
